antismash/modules/nrps_pks/test/test_smiles_generator.py

```diff
@@ -151,3 +152,28 @@ def test_ring(self):
     def test_carbon_rings(self):
         bonds = get_bond_counts(Bonds("c1cccc1"))
         assert bonds == [0, 0, 0, 0, 0]
+
+
+class TestMethylation:
+    def test_c(self):
+        front = "NC(C(=O)C)"
+        back = "C(=O)O"
+        assert methylate(front + back, "C") == front + "(C)" + back
```

Comment:
This looks wrong, we're not methylating the side-chain here.
`methylate("NC(C(=O)C)C(=O)O", "C")` returns `NC(C(=O)C)(C)C(=O)O`, which has a second side-chain at the back-bone carbon. I'd expect `NC(C(=O)C(C))C(=O)O` instead.